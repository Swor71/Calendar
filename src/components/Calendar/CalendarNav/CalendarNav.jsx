import React, { PureComponent } from 'react';

import './CalendarNav.css';

class CalendarNav extends PureComponent {
  render() {
    return (
      <div className="calendar-nav-wrapper">
        <button className="btn-arrow-left">⟵</button>
        <span className="month-year">June 2018</span>
        <button className="btn-arrow-right">⟶</button>
      </div>
    );
  }
}

export default CalendarNav;
