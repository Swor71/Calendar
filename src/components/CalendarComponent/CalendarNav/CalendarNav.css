.calendar-nav__wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

[class*='btn--'] {
  color: lightgray;
  font-size: 1.5em;
  padding: 2px 10px;
  background: transparent;
  border: 1px solid lightgray;
  border-radius: 4px;
  cursor: pointer;
}

.calendar-nav__date {
  font-weight: bold;
  font-size: 1.4em;
}
