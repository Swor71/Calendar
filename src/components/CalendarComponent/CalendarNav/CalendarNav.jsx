import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './CalendarNav.css';
import { monthList, currentMonth, currentYear } from '../../../utils/dateFormats';

class CalendarNav extends Component {
  static propTypes = {
    currentDate: PropTypes.string,
    onHandleMonthChange: PropTypes.func,
  };

  static defaultProps = {
    currentDate: `${monthList[currentMonth]} ${currentYear}`
  };

  render() {
    const { currentDate, onHandleMonthChange } = this.props;
    return (
      <div className="calendar-nav__wrapper">
        <button
          className="calendar-nav__btn--arrow-left"
          onClick={() => onHandleMonthChange(-1)}
        >⟵</button>
        <span className="calendar-nav__date">{currentDate}</span>
        <button
          className="calendar-nav__btn--arrow-right"
          onClick={() => onHandleMonthChange(1)}
        >⟶</button>
      </div>
    );
  }
}

export default CalendarNav;
