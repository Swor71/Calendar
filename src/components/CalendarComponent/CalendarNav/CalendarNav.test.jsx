import React from 'react';
import { shallow } from 'enzyme';
import CalendarNav from './CalendarNav';

describe('CalendarNav', () => {
  let consoleSpy;

  beforeAll(() => {
    consoleSpy = jest.spyOn(console, 'error');
  });

  it('should match snapshot without current date prop', () => {
    const component = shallow(<CalendarNav currentDate="January 2022" />);

    expect(component).toMatchSnapshot();
  });

  it('should render the component and verify whether arrows work properly', () => {
    const propsMock = {
      onHandleMonthChange: jest.fn(),
    };
    const component = shallow(<CalendarNav {...propsMock} />);
    const arrowLeft = component.find('.calendar-nav__btn--arrow-left');
    const arrowRight = component.find('.calendar-nav__btn--arrow-right');

    arrowLeft.simulate('click');

    expect(propsMock.onHandleMonthChange).toHaveBeenCalledTimes(1);
    expect(propsMock.onHandleMonthChange).toHaveBeenCalledWith(-1);

    arrowRight.simulate('click');

    expect(propsMock.onHandleMonthChange).toHaveBeenCalledTimes(2);
    expect(propsMock.onHandleMonthChange).toHaveBeenCalledWith(1);

    arrowLeft.simulate('click');
    arrowRight.simulate('click');
    arrowLeft.simulate('click');

    expect(propsMock.onHandleMonthChange).toHaveBeenCalledTimes(5);
  });

  it('shows no errors on console', () => {
    expect(consoleSpy).not.toHaveBeenCalled();
  });
});
