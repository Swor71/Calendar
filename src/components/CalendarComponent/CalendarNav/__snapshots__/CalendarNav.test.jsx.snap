// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarNav should match snapshot without current date prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalendarNav
    currentDate="January 2022"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="calendar-nav__btn--arrow-left"
          onClick={[Function]}
        >
          ⟵
        </button>,
        <span
          className="calendar-nav__date"
        >
          January 2022
        </span>,
        <button
          className="calendar-nav__btn--arrow-right"
          onClick={[Function]}
        >
          ⟶
        </button>,
      ],
      "className": "calendar-nav__wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "⟵",
          "className": "calendar-nav__btn--arrow-left",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "⟵",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "January 2022",
          "className": "calendar-nav__date",
        },
        "ref": null,
        "rendered": "January 2022",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "⟶",
          "className": "calendar-nav__btn--arrow-right",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "⟶",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="calendar-nav__btn--arrow-left"
            onClick={[Function]}
          >
            ⟵
          </button>,
          <span
            className="calendar-nav__date"
          >
            January 2022
          </span>,
          <button
            className="calendar-nav__btn--arrow-right"
            onClick={[Function]}
          >
            ⟶
          </button>,
        ],
        "className": "calendar-nav__wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "⟵",
            "className": "calendar-nav__btn--arrow-left",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "⟵",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "January 2022",
            "className": "calendar-nav__date",
          },
          "ref": null,
          "rendered": "January 2022",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "⟶",
            "className": "calendar-nav__btn--arrow-right",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "⟶",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
